spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/RemoveRequestParameterGatewayFilterFactory.java

```diff
@@ -57,14 +58,19 @@ public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
 				MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>(request.getQueryParams());
 				queryParams.remove(config.getName());
```

Comment:
`ServerHttpRequest.getRequest()` returns `decoded query parameters` as a response.
So, it finds the `parameter` in the `decoded query parameters` and removes them.